scripts/xdist/get_worker_test_list.py

```diff
@@ -0,0 +1,50 @@
+"""
+This script strips the console log of a pytest-xdist Jenkins run into the test
+lists of each pytest worker.
+Assumes the following format:
+[test-suite] [worker] RESULT test
+"""
+import click
+import io
+import re
+import os
+import shutil
+
+
+@click.command()
+@click.option(
+    '--log-file',
+    help="File name of console log .txt file from a Jenkins build "
+    "that ran pytest-xdist.",
+    required=True
+)
+@click.option(
+    '--test-suite',
+    help="Test suite that the pytest worker ran. Example: lms-unit",
+    required=True
+)
+def main(log_file, test_suite):
+    worker_test_dict = {}
+    with io.open(log_file, 'r') as console_file:
+        for line in console_file:
+            regex_search = re.search(r'\[lms-unit] \[gw\d+] PASSED|FAILED|SKIPPED'.format(test_suite), line)
+            if regex_search:
+                worker_num_string = "".join(re.findall('\d', regex_search.group(0)))
```

Comment:
You should be able to define a group around the ID using parentheses, so you don't need to do this secondary string search.

Reply:
Aha! Thank you... I'm a little rusty on reg ex.